edxval/api.py

```diff
@@ -968,26 +964,43 @@ def import_from_xml(xml, edx_video_id, course_id=None):
             'bitrate': encoded_video_el.get('bitrate'),
         })
     create_video(data)
-    create_transcript_objects(xml)
+    create_transcript_objects(xml, edx_video_id, resource_fs, static_dir)
 
 
-def create_transcript_objects(xml):
+def create_transcript_objects(xml, edx_video_id, resource_fs, static_dir):
     """
     Create VideoTranscript objects.
 
     Arguments:
-        xml (Element): lxml Element object
+        xml (Element): lxml Element object.
+        edx_video_id (str): Video id of the video.
+        resource_fs (OSFS): Import file system.
+        static_dir (str): The Directory to retrieve transcript file.
     """
     for transcript in xml.findall('.//transcripts/transcript'):
         try:
-            create_or_update_video_transcript(
-                transcript.attrib['video_id'],
-                transcript.attrib['language_code'],
-                metadata=dict(
-                    provider=transcript.attrib['provider'],
-                    file_name=transcript.attrib['file_name'],
-                    file_format=transcript.attrib['file_format'],
+            file_format = transcript.attrib['file_format']
+            language_code = transcript.attrib['language_code']
+            transcript_data = get_video_transcript_data(edx_video_id, language_code)
+
+            # First check if transcript record does not exist.
+            if not transcript_data:
```

Comment:
When will this exist? Is this a valid scenario ? because I guess if a video is already there then we dont do any further import.

Reply:
Yes, this is a case when we have transcripts already stored with the same language.

Reply:
Got it!